apps/etterlatte-behandling/src/main/kotlin/brev/TilbakekrevingBrevService.kt

```diff
@@ -0,0 +1,339 @@
+import java.util.UUID
+
+class TilbakekrevingBrevService(
+    val sakService: SakService,
+    val brevKlient: BrevKlient,
+    val vedtakKlient: VedtakKlient,
+    val grunnlagKlient: GrunnlagKlient,
+) {
+    suspend fun opprettVedtaksbrev(
+        behandlingId: UUID,
+        sakId: SakId,
+        bruker: BrukerTokenInfo,
+    ) {
+        val brevRequest =
+            retryOgPakkUt {
+                utledBrevRequest(bruker, behandlingId, sakId)
+            }
+
+        brevKlient.opprettVedtaksbrev(
+            behandlingId,
+            bruker,
+            brevRequest,
+        )
+    }
+
+    suspend fun genererPdf(
+        brevID: BrevID,
+        behandlingId: UUID,
+        sakId: SakId,
+        bruker: BrukerTokenInfo,
+        skalLagres: Boolean,
+    ): Pdf {
+        val brevRequest =
+            retryOgPakkUt {
+                utledBrevRequest(bruker, behandlingId, sakId, skalLagres)
+            }
+
+        return brevKlient.genererPdf(brevID, behandlingId, bruker, brevRequest)
+    }
+
+    suspend fun tilbakestillVedtaksbrev(
+        brevID: BrevID,
+        behandlingId: UUID,
+        sakId: SakId,
+        bruker: BrukerTokenInfo,
+    ): BrevPayload {
+        val brevRequest =
+            retryOgPakkUt {
+                utledBrevRequest(bruker, behandlingId, sakId)
+            }
+
+        return brevKlient.tilbakestillVedtaksbrev(
+            brevID,
+            behandlingId,
+            bruker,
+            brevRequest,
+        )
+    }
+
+    private suspend fun utledBrevRequest(
+        bruker: BrukerTokenInfo,
+        behandlingId: UUID,
+        sakId: SakId,
+        skalLagres: Boolean = false,
+    ): BrevRequest =
+        coroutineScope {
+            val sak =
+                inTransaction {
+                    sakService.finnSak(sakId) ?: throw InternfeilException("Fant ikke sak med id=$sakId")
+                }
+
+            val vedtakDeferred =
+                async {
+                    vedtakKlient.hentVedtak(behandlingId, bruker)
+                }
+            val grunnlagDef =
+                async {
+                    grunnlagKlient.hentGrunnlagForSak(sak.id, bruker) // TODO må være hentGrunnlag for vanlig behandling
+                }
+
+            // TODO trenger vi egt async?
+            val grunnlag = grunnlagDef.await()
+
+            val vedtak =
+                vedtakDeferred.await()
+                    ?: throw InternfeilException("Kan ikke lage vedtaksbrev for tilbakekreving uten vedtak behandlingId=$behandlingId")
+
+            val verge = hentVergeForSak(sak.sakType, null, grunnlag)
+            val soeker = grunnlag.mapSoeker(null)
+
+            val innloggetSaksbehandlerIdent = bruker.ident() // TODO bør ikke være nødvendig for kun vedtaksbrev?
+            val saksbehandlerIdent = vedtak.vedtakFattet?.ansvarligSaksbehandler ?: innloggetSaksbehandlerIdent
+            val attestantIdent = vedtak.attestasjon?.attestant ?: innloggetSaksbehandlerIdent
+
```

Comment:
Lurer på om utledningen her kunne vært tettere koblet på hva vi har i tilbakekrevingen / vedtaket. Støtter vi null i attestantIdent? Tror egentlig det blir "bedre" for visningen hvis saksbehandler ser på brevet og det ikke er en attestant som ser på det. Men det er nok litt utenfor POC'en